src/cljs/villagebookUI/middleware.cljs

```diff
@@ -0,0 +1,13 @@
+(ns villagebookUI.middleware
+  (:require [villagebookUI.store :as store]
+            [villagebookUI.components.utils :as utils]
+            [villagebookUI.components.login :refer [login]]))
+
+(defn protected
```

Comment:
Would `require-login` be a better name?

Reply:
Given that it's a middleware ,`require-login` conveys more meaning, yes.